src/wordpress/backup.py

```diff
@@ -142,8 +147,32 @@ def backup(self):
             self.generate_wp_files()
             self.generate_db_dump()
             logging.info("%s - %s WP backup is created", repr(self.wp_site), self.backup_pattern)
-            return True
-
+            result = True
         except WPException as err:
             logging.error("%s - WP backup failed: %s", repr(self.wp_site), err)
-            return False
+
+        if Utils.get_mandatory_env("WP_ENV") in settings.PROMETHEUS_OPENSHIFT_ENV_LIST:
+            self.prometheus_monitoring(result, self.wp_site)
+
+        return result
+
+    def prometheus_monitoring(self, backup_status, wp_site):
+
+        def my_auth_handler(url, method, timeout, headers, data):
+            username = Utils.get_mandatory_env("PROMETHEUS_PUSHGATEWAY_USERNAME")
+            password = Utils.get_mandatory_env("PROMETHEUS_PUSHGATEWAY_PASSWORD")
+            return basic_auth_handler(url, method, timeout, headers, data, username, password)
+
+        registry = CollectorRegistry()
+        if backup_status:
+            status = "backup OK"
```

Comment:
Est-ce que tu peux plutôt utiliser 1 ou 0 ? Ça permet de faire des sommes.

Par exemple, la métrique «up» (https://prometheus.io/docs/concepts/jobs_instances/#automatically-generated-labels-and-time-series) vaut toujours 1.

Reply:
Ok. C'est corrigé